app/src/main/java/com/group80/uoftinder/UserScoreComparator.java

```diff
@@ -0,0 +1,63 @@
+package com.group80.uoftinder;
+import java.util.Comparator;
+
+public class UserScoreComparator implements Comparator<Integer> {
+    private final int answerLen;
+    private final boolean[] isMultiSelect;
+    private final int[] answerBitLengths;
+    private final int scoreLength;
+
+    public UserScoreComparator(boolean[] isMultiSelect, int[] answerBitLengths) {
+        this.answerLen = isMultiSelect.length;
+        this.isMultiSelect = isMultiSelect;
+        this.answerBitLengths = answerBitLengths;
+        this.scoreLength = calculateSum(this.answerBitLengths);
+    }
+
+    private int calculateSum(int[] myArray) {
+        int total = 0;
+        for (int elem : myArray) {
+            total += elem;
+        }
+        return total;
+    }
+
+    @Override
+    public int compare(Integer score1, Integer score2) {
+        System.out.println("scoreLength: " + this.scoreLength);
+        String score1Binary = String.format("%" + this.scoreLength + "s", Integer.toBinaryString(score1)).replace(' ', '0');
+        String score2Binary = String.format("%" + this.scoreLength + "s", Integer.toBinaryString(score2)).replace(' ', '0');
+
+        System.out.println("score1Binary: " + score1Binary);
+        System.out.println("score2Binary: " + score2Binary);
```

Comment:
It's really nice that you have added these outputs for testing purposes, but please make sure to remove them afterwards :) ~

Reply:
Will do!